core/rawdb/freezer.go

```diff
@@ -164,22 +166,22 @@ func (f *freezer) Close() error {
 // in the freezer.
 func (f *freezer) HasAncient(kind string, number uint64) (bool, error) {
 	if table := f.tables[kind]; table != nil {
-		return table.has(number), nil
+		return table.has(number - f.offset), nil
 	}
 	return false, nil
 }
 
 // Ancient retrieves an ancient binary blob from the append-only immutable files.
 func (f *freezer) Ancient(kind string, number uint64) ([]byte, error) {
 	if table := f.tables[kind]; table != nil {
-		return table.Retrieve(number)
+		return table.Retrieve(number - f.offset)
 	}
 	return nil, errUnknownTable
 }
 
 // Ancients returns the length of the frozen items.
 func (f *freezer) Ancients() (uint64, error) {
```

Comment:
Please check all the usage of `Ancients`, make sure they are all right.

Reply:
Done, for the usage of `Ancients`, introduced a function for return actual number of items in current ancientDB, and in the usage area of `Ancients`, some means the actual number, some means all the frozen blocks including pruned in this feature, all handled case by case, please check.